src/nnet3/nnet-simple-component.cc

```diff
@@ -1171,6 +1171,118 @@ void RectifiedLinearComponent::StoreStats(
   StoreStatsInternal(out_value, &temp_deriv);
 }
 
+void EluComponent::Propagate(
+    const ComponentPrecomputedIndexes *indexes,
+    const CuMatrixBase<BaseFloat> &in,
+    CuMatrixBase<BaseFloat> *out) const {
+  // Apply exponential linear function (x >= 0 ? x : exp(x) - 1)
+  out->CopyFromMat(in);
+  CuMatrix<BaseFloat> exp_negated_relu(in);
```

Comment:
You can use the per-element Min() function [which was just now added], to make this more efficient. for propagate: copy in->output, exponentiate output, do output->Min(input).